hedera-node/src/main/java/com/hedera/services/records/TxnAwareRecordsHistorian.java

```diff
@@ -67,15 +71,25 @@ public class TxnAwareRecordsHistorian implements RecordsHistorian {
 	private RecordStreamObject topLevelStreamObj;
 
 	@Inject
-	public TxnAwareRecordsHistorian(RecordCache recordCache, TransactionContext txnCtx, ExpiryManager expiries) {
+	public TxnAwareRecordsHistorian(RecordCache recordCache, TransactionContext txnCtx, ExpiryManager expiries,
+			ConsensusTimeTracker consensusTimeTracker) {
 		this.expiries = expiries;
 		this.txnCtx = txnCtx;
 		this.recordCache = recordCache;
+		this.consensusTimeTracker = consensusTimeTracker;
 	}
 
 	@Override
 	public Instant nextFollowingChildConsensusTime() {
-		return txnCtx.consensusTime().plusNanos(1L + followingChildRecords.size());
+		Instant next = txnCtx.consensusTime().plusNanos(1L + followingChildRecords.size());
+
+		if (!consensusTimeTracker.isAllowableFollowingTime(next)) {
```

Comment:
Could `consensusTimeTracker` just return the next time to use here? (And throw if no usable time exists.)